Lesson16/lesson16/gokce/script.js

```diff
@@ -0,0 +1,388 @@
+    - Define a function `printObjectKeys(obj)` that uses a for-in loop
+      to log each key of the object.
+    - Example: { name: "Alice", age: 25 } -> logs "name", then "age"
+*/
+function printObjectKeys(obj) {
+  for (const key in obj) {
+    console.log(key);
+  }
+}
+
+printObjectKeys(person);
+/*
+11. Sum Array Using do-while Loop
+    - Define a function `sumWithDoWhile(numbers)` that uses a do-while loop
+      to sum all numbers in the array.
+    - Log the total.
+*/
+function sumWithDoWhile(numbers) {
+  let total = 0;
+  let index = 0;
+  do {
+    total += numbers([index]);
+    console.log(total);
+  } while (index < numbers.length);
+}
+sumWithDoWhile(numbers);
+sumWithDoWhile([1, 2, 3, 4, 5]);
+
+/*
+12. Remove Duplicates from an Array
+    - Define a function `removeDuplicates(arr)` that loops through the array
+      and creates a new array without duplicate elements.
+    - Hint: you could check if the item is already in the new array before pushing.
+    - Log the new array without duplicates.
+*/
+
+const ex1Array = [1, 2, 2, 3, 4, 4, 5];
+function removeDuplicates(arr) {
+  let removeDup = [];
+  for (let i = 0; i < arr.length; i++) {
+    if (removeDup.indexOf(arr[i] === -1)) {
+      removeDup.push(arr[i]);
+    }
+    console.log(removeDup);
+  }
+}
+removeDuplicates(ex1Array);
+
+/*
+13. Calculate Factorial (For Loop)
+    - Define a function `factorial(n)` that calculates n! (n factorial)
+      using a for loop.
+    - Log the result. 
+    - Example: factorial(5) -> 120
+*/
+function factorial(n) {
+  let result = 1;
+  for (let i = n; i >= 1; i--) {
+    result = result * i;
+  }
+  console.log(result);
+}
+factorial(5);
+
+/*
+14. String -> Array -> String
+    - Define a function `reverseWords(sentence)` that splits the sentence 
+      into an array of words, reverses the array order, then joins it back into
+      a string. Use loops or built-in methods as you like.
+    - Log the reversed sentence.
+*/
+function reverseWords(sentence) {
+  let reversedSentence = [];
+  reversedSentence = sentence.split(" ").reverse().join(" ");
+  console.log(reversedSentence);
+}
+
+reverseWords("Life is challenging sometimes.");
+/*
+15. Filter Words Longer Than X
+    - Define a function `filterLongWords(words, minLength)` that uses a for loop
+      to collect only the words that have a length >= minLength.
+    - Log the resulting array.
+*/
+function filterLongWords(words, minLength) {
+  let resultArray = [];
+  for (let i = 0; i < words.length; i++) {
+    if (words[i].length >= minLength) {
+      resultArray.push(words[i]);
+    }
+  }
+  console.log(resultArray);
+}
+filterLongWords(["apple", "banana", "cherry", "kiwi"], 6);
+
+/*
+16. Log Array Elements with Their Indices
+    - Define a function `logElementsWithIndex(arr)` that loops through the array
+      and logs "Index: i, Value: arr[i]" for each element.
+*/
+
+function logElementsWithIndex(arr) {
+  for (const element of arr) {
```

Comment:
For of loop gives you the element directly with no index. Instead, you need to use the standard for loop here:
```suggestion
  for (let i=0; i<arr.length; i++) {
```